pypeman/plugins/remoteadmin/views.py

```diff
@@ -112,8 +114,11 @@ async def list_msgs(request, ws=None):
         text=text, rtext=rtext) or []
 
     for res in messages:
+        res["id"] = res["id"]
+        res["state"] = res.get("state", "UNKNOWN")
         res['timestamp'] = res['message'].timestamp_str()
-        res['message'] = res['message'].to_json()
+        if "message" in res:
+            res.pop("message")
```

Comment:
is message_store.search always returning a copy?

Or are some un memory implementations returning the stored dict.
If this might be the case, then pop will destroy the message. if you don't create a copy of the dict. before popping.

Reply:
Ohhhh, yeah, good question ! You made me doubt ! 
I searched and no, the MemoryMessageStore is intelligent and create new messages Objects so i cannot delete messsages without calling the delete func of the store.